src/service/ampdoc-impl.js

```diff
@@ -120,6 +120,14 @@ export class AmpDocService {
         }
       }
 
+      // Traverse the boundary of a friendly iframe.
+      const win = (n.ownerDocument || n).defaultView;
+      if (win && win != this.win && win.frameElement) {
```

Comment:
Yeah, but you want to avoid exceptions.

On Tue, Sep 13, 2016 at 8:59 AM, Dima Voytenko <notifications@github.com>
wrote:

> In src/service/ampdoc-impl.js
> <https://github.com/ampproject/amphtml/pull/4916#discussion_r78589724>:
>
> > @@ -120,6 +120,14 @@ export class AmpDocService {
> >          }
> >        }
> >
> > +      // Traverse the boundary of a friendly iframe.
> > +      const win = (n.ownerDocument || n).defaultView;
> > +      if (win && win != this.win && win.frameElement) {
>
> It's null. But also I assume that the only way I'd get access to this
> element is in the same-origin case.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4916/files/7044c7527ca7e8a973cf368d8caf77f50a31da06#r78589724>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFeT6GUowhY7MMDzSsjF17iuHrXjZTEks5qpshagaJpZM4J5k6k>
> .
>